applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.cpp

```diff
@@ -133,16 +138,16 @@ void ImposeRigidMovementProcess::ExecuteInitialize()
     const SizeType number_of_double_variables = master_double_list_variables.size();
     const SizeType number_of_components_variables = master_components_list_variables.size();
 
-    // Creating a buffer for parallel vector fill
-    const int num_threads = OpenMPUtils::GetNumThreads();
-    std::vector<ConstraintContainerType> constraints_buffer(num_threads);
+    // We get the relation and constant
+    const double relation = mThisParameters["relation"].GetDouble();
+    const double constant = mThisParameters["constant"].GetDouble();
 
     // Reference constraint
     const auto& r_clone_constraint = KratosComponents<MasterSlaveConstraint>::Get("LinearMasterSlaveConstraint");
 
-    #pragma omp parallel
+    #pragma omp parallel firstprivate(relation, constant)
```

Comment:
is it necessary to make them firstprivate?

Reply:
I suppose not, because is constat, this makes a copy of a constant value